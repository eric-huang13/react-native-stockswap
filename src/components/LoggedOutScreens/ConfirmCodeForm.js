import React, {useState} from 'react';
import {SafeAreaView, Text, View, StyleSheet, KeyboardAvoidingView, TouchableOpacity} from 'react-native';
import {moderateScale} from '../../util/responsiveFont';
import LinearGradient from 'react-native-linear-gradient';
import SmallStockSwap from '../../icons/SmallStockSwap';

import {
  CodeField,
  Cursor,
  useBlurOnFulfill,
  useClearByFocusCell,
} from 'react-native-confirmation-code-field';

const styles = StyleSheet.create({
  root: {flex: 1, padding: 20},
  title: {textAlign: 'center', fontSize: 30},
  codeFieldRoot: {marginTop: 20},
  cell: {
    width: 42,
    height: 42,
    lineHeight: 38,
    fontSize: 24,
    borderWidth: 2.1,
    borderColor: '#FFFFFF',
    textAlign: 'center',
    backgroundColor:'#FFFFFF'
  },
  focusCell: {
    borderColor: '#B8A0FF',
  },
});

const CELL_COUNT = 6;

const ConfirmCodeForm = ({navigation, email, value, setValue}) => {
  // const [value, setValue] = useState('');
  const ref = useBlurOnFulfill({value, cellCount: CELL_COUNT});
  const [props, getCellOnLayoutHandler] = useClearByFocusCell({
    value,
    setValue,
  });
  console.log(email,"PROPSs")

  return (
    <LinearGradient
    start={{x: 0.1, y: 1}}
    end={{x: 0.1, y: 0.1}}
    colors={['#1D2842', '#3d4b6e']}
    style={{flex: 1}}>
    <KeyboardAvoidingView
      behavior={Platform.OS === 'ios' ? 'padding' : null}
      style={{flex: 1}}>
    <SafeAreaView style={styles.root}>
    <View style={style.inner}>

    <View style={style.stockHeader}>
                        <SmallStockSwap />
                      </View>
                      <View style={style.container}>
                      <Text style={style.welcomeHeader}>Verification Code</Text>
      <CodeField
        ref={ref}
        {...props}
        // Use `caretHidden={false}` when users can't paste a text value, because context menu doesn't appear
        value={value}
        onChangeText={setValue}
        cellCount={CELL_COUNT}
        rootStyle={styles.codeFieldRoot}
        keyboardType="number-pad"
        textContentType="oneTimeCode"
        renderCell={({index, symbol, isFocused}) => (
          <Text
            key={index}
            style={[styles.cell, isFocused && styles.focusCell]}
            onLayout={getCellOnLayoutHandler(index)}>
            {symbol || (isFocused ? <Cursor /> : null)}
          </Text>
        )}
      />
 <View style={style.termsContainer}>
                          <View style={style.leftTerms}>
                            <Text style={style.newText}>Need a new code?</Text>
                            <TouchableOpacity
                              onPress={() =>
                                this.props.navigation.navigate('SignUp')
                              }>
                              <Text style={style.termsText}>Resend Code</Text>
                            </TouchableOpacity>
                          </View>
                          <TouchableOpacity
                            onPress={() =>
                             navigation.navigate('Login')
                            }>
                            <Text style={style.termsText}>Back to Login</Text>
                          </TouchableOpacity>
                        </View>
<View  style={style.buttonView}>

      <TouchableOpacity onPress={() => navigation.navigate({
                                name: 'NewPassword',
                                params: {codeInput: value, email:email},
                              }) }>
                            <Text style={style.button}>Send Reset Link</Text>
                          </TouchableOpacity>
                          
                          </View>
                          </View>
                          </View>

                          
    </SafeAreaView>
    </KeyboardAvoidingView>
      </LinearGradient>
  );
};

export default ConfirmCodeForm;

const style = StyleSheet.create({
    mainContainer: {
      flex: 1,
      padding: moderateScale(8),
      paddingHorizontal: moderateScale(24),
    },
    inner: {
      justifyContent: 'flex-end',
    },
    stockHeader: {
      flexDirection: 'row',
      justifyContent: 'center',
      marginTop: moderateScale(20),
      marginBottom: moderateScale(36),
    },
    stockText: {
      fontSize: moderateScale(27),
      fontWeight: 'bold',
      color: 'white',
    },
    swapText: {
      fontSize: moderateScale(27),
      fontWeight: 'bold',
      color: '#b8a0ff',
    },
    container: {
      borderRadius: moderateScale(16),
      backgroundColor: '#303e5e',
      paddingHorizontal: moderateScale(20),
      paddingVertical: moderateScale(20),
      flexDirection: 'column',
      shadowColor: 'rgba(0,0,0,0.13)',
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowOpacity: 0.25,
      shadowRadius: 3.84,
  
      elevation: 1,
    },
    welcomeHeader: {
      color: '#FFFFFF',
      fontSize: moderateScale(22),
      marginBottom: moderateScale(8),
      fontFamily: 'Montserrat-Bold',
      textAlign:'center',
    },
    loginHeader: {
      color: '#9299a9',
      marginBottom: moderateScale(18),
      fontSize: moderateScale(15),
      fontFamily: 'Montserrat-Medium',
    },
    inputHeader: {
      fontSize: moderateScale(14),
      color: '#babec8',
      marginBottom: 1,
      fontFamily: 'Montserrat-Regular',
    },
    inputStyle: {
      borderRadius: moderateScale(8),
      padding: moderateScale(8),
      marginTop: moderateScale(1),
      fontSize: moderateScale(16),
      fontFamily: 'Montserrat-Italic',
      backgroundColor: '#536183',
      opacity: 0.7,
      color: '#FFFFFF',
    },
    inputStyleConfirm: {
      borderRadius: moderateScale(8),
      marginBottom: moderateScale(12),
      padding: moderateScale(8),
      marginTop: moderateScale(1),
      fontSize: moderateScale(16),
      fontFamily: 'Montserrat-Italic',
      backgroundColor: '#536183',
      opacity: 0.7,
      color: '#9ea6b5',
    },
    buttonView: {
        borderRadius: moderateScale(16),

        flexDirection: 'row',
        justifyContent: 'center',
        marginTop: moderateScale(20),    },
    button: {
      alignSelf: 'center',
      backgroundColor: '#8B64FF',
      color: '#FFFFFF',
      textAlign: 'center',
      paddingVertical: moderateScale(12),
      paddingHorizontal: moderateScale(8),
      width: moderateScale(162),
      borderRadius: moderateScale(6),
      fontSize: moderateScale(16),
      fontFamily: 'Montserrat-SemiBold',
    },
    termsContainer: {
      marginTop: moderateScale(28),
      flexDirection: 'row',
      justifyContent: 'space-between',
      paddingHorizontal: moderateScale(2),
    },
    leftTerms: {
      flexDirection: 'row',
    },
    newText: {
      color: '#FFFFFF',
      fontSize: moderateScale(12),
      marginRight: moderateScale(3),
      fontFamily: 'Montserrat-Medium',
    },
    termsText: {
      color: '#B8A0FF',
      fontSize: moderateScale(12),
      fontFamily: 'Montserrat-Medium',
    },
    bottomButtonsContainer: {
      alignItems: 'center',
    },
    orText: {
      marginVertical: moderateScale(16),
      color: '#CBCDD7',
      fontSize: moderateScale(14),
      fontFamily: 'Montserrat-Regular',
    },
    alternateSignUpContainer: {
      flexDirection: 'column',
      justifyContent: 'space-between',
    },
    alternateSignUpButton: {
      alignSelf: 'center',
      justifyContent: 'center',
      color: '#FFFFFF',
      textAlign: 'center',
      fontSize: moderateScale(14),
      fontFamily: 'Montserrat-SemiBold',
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowOpacity: 0.25,
      shadowRadius: 3.84,
      elevation: 1,
    },
    alternateSignupInner: {
      alignSelf: 'center',
      paddingVertical: moderateScale(12),
      paddingHorizontal: moderateScale(4),
      backgroundColor: '#2C3957',
      width: moderateScale(350),
      borderRadius: moderateScale(8),
      marginBottom: moderateScale(10),
      flexDirection: 'row',
    },
    signupIcon: {
      padding: moderateScale(7),
      backgroundColor: '#3A4A6D',
      borderRadius: moderateScale(7),
      marginVertical: moderateScale(-8),
      marginRight: moderateScale(63),
      alignSelf: 'center',
    },
    errorText: {
      color: '#F66E6E',
      fontWeight: 'bold',
      marginBottom: moderateScale(10),
      marginTop: moderateScale(1),
      textAlign: 'center',
    },
  });
  