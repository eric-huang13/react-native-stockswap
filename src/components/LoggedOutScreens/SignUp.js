import React, {useState, useEffect} from 'react';
import {
  TextInput,
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  KeyboardAvoidingView,
  SafeAreaView,
  TouchableWithoutFeedback,
  Keyboard,
  ScrollView,
  Modal,
  Platform,
} from 'react-native';
import {Formik} from 'formik';
import * as yup from 'yup';
import {connect} from 'react-redux';
import {Register} from '../../actions/user';
import {RegisterGoogleSignup} from '../../actions/user';

import CheckBox from '@react-native-community/checkbox';
import LinearGradient from 'react-native-linear-gradient';
import SmallStockSwap from '../../icons/SmallStockSwap';
import AppleIcon from '../../icons/AppleIcon';
import FacebookIcon from '../../icons/FacebookIcon';
import TermsAndConditions from './TermsAndConditions';
import GoogleOauth from '../LoggedOutComponents/GoogleOauth';

import {moderateScale} from '../../util/responsiveFont';

import Toast from 'react-native-toast-message';

const reviewSchema = yup.object({
  email: yup
    .string()
    .required('Email is required')
    .email('A valid email address is required'),

  password: yup
    .string()
    .required('Password is required')
    .min(8, ({min}) => `Password must be at least ${min} characters`)
    .matches(/\d/, 'Password must have a number')
    .matches(/\w*[a-z]\w*/, 'Password must have a lowercase letter')
    .matches(/\w*[A-Z]\w*/, 'Password must have a capital letter')
    .matches(
      /[!@#$%^&*()\-_"=+{}; :,<.>]/,
      'Password must have a special character',
    ),

  passwordConfirmation: yup
    .string()
    .required('Password confimation is required')
    .oneOf([yup.ref('password'), null], 'Passwords must match'),

  termsVersion: yup.string().required('Please agree with Terms and Conditions'),
});

export const SignUp = ({
  RegisterUser,
  navigation,
  userData,
  loading,
  RegisterUserGoogle,
}) => {
  const [toggleCheckBox, setToggleCheckBox] = useState(false);
  const [termsModal, setTermsModal] = useState(false);

  //modal
  const handleTerms = (item) => {
    setTermsModal(item);
  };

  //toggle check
  const handleCheck = (item) => {
    setToggleCheckBox(item);
  };
  useEffect(() => {
    loading === true
      ? Toast.show({
          type: 'info',
          text2: 'Sending credentials...',
        })
      : null;
  }, [loading]);

  return (
    <LinearGradient
      start={{x: 0.1, y: 1}}
      end={{x: 0.1, y: 0.1}}
      colors={['#1D2842', '#3d4b6e']}
      style={{flex: 1}}>
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : null}
        style={{flex: 1}}>
        <SafeAreaView style={style.mainContainer}>
          <ScrollView>
            <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
              <Formik
                initialValues={{
                  email: '',
                  password: '',
                  passwordConfirmation: '',
                  termsVersion: '',
                }}
                validationSchema={reviewSchema}
                onSubmit={(values, actions) => {
                  console.log(values, 'Values');
                  // navigation.navigate({
                  //   name: 'ProfileInfoForm',
                  //   params: {userInfo: values},
                  // });
                  RegisterUser({
                    email: values.email,
                    password: values.password,
                  });
                }}>
                {(props) => (
                  <View style={style.inner}>
                    <Modal
                      transparent={true}
                      visible={termsModal}
                      animationType="slide">
                      <TermsAndConditions
                        handleTerms={handleTerms}
                        handleCheck={handleCheck}
                        props={props}
                        navigation={navigation}
                      />
                    </Modal>
                    <View style={style.stockHeader}>
                      <SmallStockSwap />
                    </View>
                    <View style={style.container}>
                      <Text style={style.signUpHeader}>Sign Up</Text>
                      <View>
                        <Text style={style.inputHeader}>Email</Text>
                        <TextInput
                          style={
                            props.touched.email && props.errors.email
                              ? {
                                  ...style.inputStyle,
                                  backgroundColor: '#F66E6E',
                                }
                              : {...style.inputStyle}
                          }
                          placeholder="Enter your email"
                          placeholderTextColor="#9ea6b5"
                          onChangeText={props.handleChange('email')}
                          onBlur={props.handleBlur('email')}
                          value={props.values.email}
                          keyboardType="email-address"
                          autoCapitalize="none"
                          returnKeyType="next"
                        />
                        <Text style={style.errorText}>
                          {props.touched.email && props.errors.email}
                        </Text>
                      </View>
                      <View>
                        <Text style={style.inputHeader}>Password</Text>

                        <TextInput
                          style={
                            props.touched.password && props.errors.password
                              ? {
                                  ...style.inputStyle,
                                  backgroundColor: '#F66E6E',
                                }
                              : {...style.inputStyle}
                          }
                          placeholder="Password"
                          onChangeText={props.handleChange('password')}
                          onBlur={props.handleBlur('password')}
                          value={props.values.password}
                          placeholder="Enter your password"
                          placeholderTextColor="#9ea6b5"
                          secureTextEntry
                          returnKeyType="next"
                        />
                      </View>
                      <Text style={style.errorText}>
                        {props.touched.password && props.errors.password}
                      </Text>
                      <View>
                        <Text style={style.inputHeader}>Repeat password</Text>

                        <TextInput
                          style={
                            props.touched.passwordConfirmation &&
                            props.errors.passwordConfirmation
                              ? {
                                  ...style.inputStyleConfirm,
                                  backgroundColor: '#F66E6E',
                                }
                              : {...style.inputStyleConfirm}
                          }
                          textContentType="password"
                          placeholder="Confirm password"
                          onChangeText={props.handleChange(
                            'passwordConfirmation',
                          )}
                          onBlur={props.handleBlur('passwordConfirmation')}
                          value={props.values.passwordConfirmation}
                          placeholder="Enter your password"
                          placeholderTextColor="#9ea6b5"
                          secureTextEntry
                        />
                        <Text style={style.errorText}>
                          {props.touched.passwordConfirmation &&
                            props.errors.passwordConfirmation}
                        </Text>
                      </View>
                      <View style={style.termsOuterContainer}>
                        <View style={style.termsInnerContainer}>
                          <CheckBox
                            style={style.checkbox}
                            disabled={true}
                            boxType="square"
                            value={toggleCheckBox}
                            onValueChange={(newValue) =>
                              setToggleCheckBox(newValue)
                            }
                            tintColors={{
                              true: '#b8a0ff',
                              false: 'lightgrey',
                            }}
                            onCheckColor="#b8a0ff"
                          />
                          <TouchableOpacity
                            style={style.termsTouchOpacity}
                            onPress={() => handleTerms(true)}>
                            <Text style={style.termsText}>
                              I agree with the Terms and Conditions
                            </Text>
                          </TouchableOpacity>
                          <Text style={style.errorText} />
                        </View>
                        <TouchableOpacity
                          onPress={() => navigation.navigate('Login')}>
                          <Text style={style.termsText}>Login</Text>
                        </TouchableOpacity>
                      </View>

                      <View style={style.termsError}>
                        <Text style={style.errorText}>
                          {props.touched.termsVersion &&
                            props.errors.termsVersion}
                        </Text>
                      </View>
                      <TouchableOpacity onPress={props.handleSubmit}>
                        <Text style={style.button}>Sign Up</Text>
                      </TouchableOpacity>
                    </View>
                  </View>
                )}
              </Formik>
            </TouchableWithoutFeedback>
            <View style={style.bottomButtonsContainer}>
              <Text style={style.orText}>--OR--</Text>
              <View style={style.alternateSignUpContainer}>
                <GoogleOauth
                  signup={'signup'}
                  RegisterUserGoogle={RegisterUserGoogle}
                />
                {/* <View style={style.alternateSignupInner}>
                  <View style={style.signupIcon}>
                    <FacebookIcon />
                  </View>
                  <Text style={style.alternateSignUpButton}>
                    LOGIN WITH FACEBOOK
                  </Text>
                </View>
                <View style={style.alternateSignupInner}>
                  <View style={style.signupIcon}>
                    <AppleIcon />
                  </View>
                  <Text style={style.alternateSignUpButton}>
                    LOGIN WITH APPLE
                  </Text>
                          </View> */}
              </View>
            </View>
          </ScrollView>
        </SafeAreaView>
      </KeyboardAvoidingView>
    </LinearGradient>
  );
};
const mapStateToProps = (state) => {
  return {
    user: state.user,
    userData: state.user.userData,
    loading: state.user.loading,
    error: state.user.error,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    RegisterUser: (input) => dispatch(Register(input)),
    RegisterUserGoogle: (input) => dispatch(RegisterGoogleSignup(input)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);

const style = StyleSheet.create({
  mainContainer: {
    flex: 1,
    padding: moderateScale(8),
    paddingHorizontal: moderateScale(24),
  },
  inner: {
    justifyContent: 'flex-end',
  },
  stockHeader: {
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: moderateScale(2),
    marginBottom: moderateScale(6),
  },
  stockText: {
    fontSize: moderateScale(27),
    fontWeight: 'bold',
    color: 'white',
  },
  swapText: {
    fontSize: moderateScale(27),
    fontWeight: 'bold',
    color: '#b8a0ff',
  },
  container: {
    borderRadius: moderateScale(8),
    backgroundColor: '#303e5e',
    paddingHorizontal: moderateScale(20),
    paddingVertical: moderateScale(18),
    flexDirection: 'column',
    shadowColor: 'rgba(0,0,0,0.13)',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,

    elevation: 1,
  },
  signUpHeader: {
    color: '#FFFFFF',
    fontSize: moderateScale(22),
    fontFamily: 'Montserrat-Bold',
    marginBottom: moderateScale(18),
  },
  inputHeader: {
    fontSize: moderateScale(14),
    color: '#babec8',
    marginBottom: moderateScale(1),
    fontFamily: 'Montserrat-Regular',
  },
  inputStyle: {
    borderRadius: moderateScale(8),
    padding: moderateScale(8),
    marginTop: moderateScale(1),
    fontSize: moderateScale(16),
    fontFamily: 'Montserrat-Italic',
    backgroundColor: '#536183',
    opacity: 0.7,
    color: '#FFFFFF',
  },
  inputStyleConfirm: {
    borderRadius: moderateScale(8),
    padding: moderateScale(8),
    marginTop: moderateScale(1),
    fontSize: moderateScale(16),
    fontFamily: 'Montserrat-Italic',
    backgroundColor: '#536183',
    opacity: 0.7,
    color: '#FFFFFF',
  },

  button: {
    alignSelf: 'center',
    backgroundColor: '#8B64FF',
    color: 'white',
    textAlign: 'center',
    paddingVertical: moderateScale(12),
    paddingHorizontal: moderateScale(20),
    width: moderateScale(162),
    borderRadius: moderateScale(6),
    fontSize: moderateScale(16),
    fontFamily: 'Montserrat-SemiBold',
  },
  termsOuterContainer: {
    marginBottom: moderateScale(0),
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: moderateScale(2),
    alignItems: 'center',
  },
  termsInnerContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  termsText: {
    color: '#b8a0ff',
    fontSize: moderateScale(11),
    fontFamily: 'Montserrat-Medium',
  },
  termsError: {
    marginBottom: moderateScale(18),
  },
  termsTouchOpacity: {
    paddingLeft: 10,
  },
  checkbox: {},
  bottomButtonsContainer: {
    alignItems: 'center',
  },
  orText: {
    marginVertical: moderateScale(16),
    color: '#CBCDD7',
    fontSize: moderateScale(14),
    fontFamily: 'Montserrat-Regular',
  },
  alternateSignUpContainer: {
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
  alternateSignUpButton: {
    alignSelf: 'center',
    justifyContent: 'center',
    color: '#FFFFFF',
    textAlign: 'center',
    fontSize: moderateScale(14),
    fontFamily: 'Montserrat-SemiBold',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 1,
  },
  alternateSignupInner: {
    alignSelf: 'center',
    paddingVertical: moderateScale(12),
    paddingHorizontal: moderateScale(4),
    backgroundColor: '#2C3957',
    width: moderateScale(350),
    borderRadius: moderateScale(8),
    marginBottom: moderateScale(10),
    flexDirection: 'row',
  },
  signupIcon: {
    padding: moderateScale(7),
    backgroundColor: '#3A4A6D',
    borderRadius: moderateScale(7),
    marginVertical: moderateScale(-8),
    marginRight: moderateScale(63),
    alignSelf: 'center',
  },
  errorText: {
    color: '#F66E6E',
    fontWeight: 'bold',
    marginBottom: moderateScale(1),
    marginTop: moderateScale(1),
    textAlign: 'center',
  },
});
