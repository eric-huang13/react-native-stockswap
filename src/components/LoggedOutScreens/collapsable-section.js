import React, {useEffect, useState} from 'react';
import {Animated, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {moderateScale} from '../../util/responsiveFont';
import CircleUp from '../../icons/arrow-circle-up-solid';

const CollapsableSection = ({title, children}) => {
  const [rotateAnimation, setRotateAnimation] = useState(new Animated.Value(0));
  const interpolateRotating = rotateAnimation.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '90deg'],
  });
  const [expanded, setExpanded] = useState(false);
  useEffect(() => {
    Animated.timing(rotateAnimation, {
      toValue: 1,
      duration: 100,
      useNativeDriver: true,
    }).start(() => {
      rotateAnimation.setValue(expanded ? 0 : 1);
    });
  }, [expanded]);

  return (
    <View style={style.section}>
      <TouchableOpacity
        style={style.touchable}
        onPress={() => setExpanded(!expanded)}>
        <Animated.View style={{transform: [{rotate: interpolateRotating}]}}>
          <CircleUp style={{padding: 5}} />
        </Animated.View>
        <Text style={style.header}>{title}</Text>
      </TouchableOpacity>
      {expanded ? (
        <Animated.View style={[{paddingVertical: 5}]}>{children}</Animated.View>
      ) : null}
    </View>
  );
};
const style = StyleSheet.create({
  section: {
    borderTopWidth: 2,
    borderColor: 'lightgrey',
    marginVertical: 5,
  },
  touchable: {
    flexDirection: 'row',
    paddingVertical: 5,
  },
  header: {
    paddingVertical: 5,
    color: 'white',
    fontSize: moderateScale(16),
    fontFamily: 'Montserrat-Bold',
  },
});

export default CollapsableSection;
