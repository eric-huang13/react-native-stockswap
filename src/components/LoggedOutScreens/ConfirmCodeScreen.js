import React, {useState} from 'react';
import {View, Text} from 'react-native';
import ConfirmCodeForm from './ConfirmCodeForm';

const ConfirmCodeScreen = ({navigation, route}) => {
  const [value, setValue] = useState('');
  return (
    <ConfirmCodeForm
      navigation={navigation}
      value={value}
      setValue={setValue}
      email={route.params.email}
    />
  );
};

export default ConfirmCodeScreen;
