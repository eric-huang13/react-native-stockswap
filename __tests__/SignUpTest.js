import React from 'react';
import renderer from 'react-test-renderer';
import {SignUp} from '../src/components/LoggedOutScreens/SignUp'
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6




// jest.mock('@react-native-community/google-signin', () => {});

// jest.mock('config/google_signin', () => ({
//     GoogleSignin: jest.fn(),
//     GoogleSigninButton: jest.requireActual(
//       '@react-native-community/google-signin'
//     ).GoogleSigninButton,
//   }))



jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
test('renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<SignUp/>);
  const tree = renderer.getRenderOutput();
  // const tree = renderer.create(<SignUp/>).toJSON();
  expect(tree).toMatchSnapshot();
});