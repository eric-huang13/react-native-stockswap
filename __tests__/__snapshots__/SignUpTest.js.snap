// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<LinearGradient
  colors={
    Array [
      "#1D2842",
      "#3d4b6e",
    ]
  }
  end={
    Object {
      "x": 0.1,
      "y": 0.1,
    }
  }
  start={
    Object {
      "x": 0.1,
      "y": 1,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <KeyboardAvoidingView
    behavior="padding"
    enabled={true}
    keyboardVerticalOffset={0}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <ForwardRef(SafeAreaView)
      style={
        Object {
          "flex": 1,
          "padding": 11.299270072992702,
          "paddingHorizontal": 33.8978102189781,
        }
      }
    >
      <ScrollView>
        <TouchableWithoutFeedback
          onPress={[Function]}
        >
          <Formik
            initialValues={
              Object {
                "email": "",
                "password": "",
                "passwordConfirmation": "",
                "termsVersion": "",
              }
            }
            onSubmit={[Function]}
            validationSchema={
              ObjectSchema {
                "_blacklist": ReferenceSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_excludedEdges": Array [],
                "_mutate": undefined,
                "_nodes": Array [
                  "termsVersion",
                  "passwordConfirmation",
                  "password",
                  "email",
                ],
                "_sortErrors": [Function],
                "_typeError": [Function],
                "_whitelist": ReferenceSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "conditions": Array [],
                "deps": Array [],
                "exclusiveTests": Object {},
                "fields": Object {
                  "email": StringSchema {
                    "_blacklist": ReferenceSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_blacklistError": undefined,
                    "_typeError": [Function],
                    "_whitelist": ReferenceSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_whitelistError": undefined,
                    "conditions": Array [],
                    "deps": Array [],
                    "exclusiveTests": Object {
                      "email": false,
                      "required": true,
                    },
                    "spec": Object {
                      "abortEarly": true,
                      "label": undefined,
                      "meta": undefined,
                      "nullable": false,
                      "presence": "required",
                      "recursive": true,
                      "strict": false,
                      "strip": false,
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "string",
                  },
                  "password": StringSchema {
                    "_blacklist": ReferenceSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_blacklistError": undefined,
                    "_typeError": [Function],
                    "_whitelist": ReferenceSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_whitelistError": undefined,
                    "conditions": Array [],
                    "deps": Array [],
                    "exclusiveTests": Object {
                      "matches": false,
                      "min": true,
                      "required": true,
                    },
                    "spec": Object {
                      "abortEarly": true,
                      "label": undefined,
                      "meta": undefined,
                      "nullable": false,
                      "presence": "required",
                      "recursive": true,
                      "strict": false,
                      "strip": false,
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "string",
                  },
                  "passwordConfirmation": StringSchema {
                    "_blacklist": ReferenceSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_blacklistError": undefined,
                    "_typeError": [Function],
                    "_whitelist": ReferenceSet {
                      "list": Set {
                        null,
                      },
                      "refs": Map {
                        "password" => Reference {
                          "getter": [Function],
                          "isContext": false,
                          "isSibling": true,
                          "isValue": false,
                          "key": "password",
                          "map": undefined,
                          "path": "password",
                        },
                      },
                    },
                    "_whitelistError": [Function],
                    "conditions": Array [],
                    "deps": Array [],
                    "exclusiveTests": Object {
                      "required": true,
                    },
                    "spec": Object {
                      "abortEarly": true,
                      "label": undefined,
                      "meta": undefined,
                      "nullable": false,
                      "presence": "required",
                      "recursive": true,
                      "strict": false,
                      "strip": false,
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "string",
                  },
                  "termsVersion": StringSchema {
                    "_blacklist": ReferenceSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_blacklistError": undefined,
                    "_mutate": undefined,
                    "_typeError": [Function],
                    "_whitelist": ReferenceSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_whitelistError": undefined,
                    "conditions": Array [],
                    "deps": Array [],
                    "exclusiveTests": Object {
                      "required": true,
                    },
                    "spec": Object {
                      "abortEarly": true,
                      "label": undefined,
                      "meta": undefined,
                      "nullable": false,
                      "presence": "required",
                      "recursive": true,
                      "strict": false,
                      "strip": false,
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "string",
                  },
                },
                "spec": Object {
                  "abortEarly": true,
                  "label": undefined,
                  "meta": undefined,
                  "nullable": false,
                  "presence": "optional",
                  "recursive": true,
                  "strict": false,
                  "strip": false,
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
                "type": "object",
              }
            }
          >
            [Function]
          </Formik>
        </TouchableWithoutFeedback>
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#CBCDD7",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 19.77372262773723,
                "marginVertical": 22.598540145985403,
              }
            }
          >
            --OR--
          </Text>
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "space-between",
              }
            }
          >
            <Memo(Connect(GoogleOauth)) />
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#2C3957",
                  "borderRadius": 11.299270072992702,
                  "flexDirection": "row",
                  "marginBottom": 14.124087591240876,
                  "paddingHorizontal": 5.649635036496351,
                  "paddingVertical": 16.94890510948905,
                  "width": 494.3430656934307,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#3A4A6D",
                    "borderRadius": 9.886861313868614,
                    "marginRight": 88.98175182481751,
                    "marginVertical": -11.299270072992702,
                    "padding": 9.886861313868614,
                  }
                }
              >
                <FacebookIcon />
              </View>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#FFFFFF",
                    "elevation": 1,
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 19.77372262773723,
                    "justifyContent": "center",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 3.84,
                    "textAlign": "center",
                  }
                }
              >
                LOGIN WITH FACEBOOK
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#2C3957",
                  "borderRadius": 11.299270072992702,
                  "flexDirection": "row",
                  "marginBottom": 14.124087591240876,
                  "paddingHorizontal": 5.649635036496351,
                  "paddingVertical": 16.94890510948905,
                  "width": 494.3430656934307,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#3A4A6D",
                    "borderRadius": 9.886861313868614,
                    "marginRight": 88.98175182481751,
                    "marginVertical": -11.299270072992702,
                    "padding": 9.886861313868614,
                  }
                }
              >
                <AppleIcon />
              </View>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#FFFFFF",
                    "elevation": 1,
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 19.77372262773723,
                    "justifyContent": "center",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.25,
                    "shadowRadius": 3.84,
                    "textAlign": "center",
                  }
                }
              >
                LOGIN WITH APPLE
              </Text>
            </View>
          </View>
        </View>
      </ScrollView>
    </ForwardRef(SafeAreaView)>
  </KeyboardAvoidingView>
</LinearGradient>
`;
